Formula/dotnet.rb

```diff
@@ -24,7 +24,22 @@ class Dotnet < Formula
   depends_on "icu4c"
   depends_on "openssl@1.1"
 
+  uses_from_macos "krb5"
+  uses_from_macos "zlib"
+
+  on_linux do
+    depends_on "llvm" => [:build, :test]
+    depends_on "libunwind"
+    depends_on "lttng-ust"
+  end
+
+  fails_with :gcc
```

Comment:
Didn't really test LLVM vs. GCC for build time support. Mainly saw build failed on GCC 5 and that Arch & Fedora have `clang`/`llvm` build dependencies, so just switched to LLVM. 

Also, not sure of LLVM as test dependency. Just added since Homebrew test automatically fails with complaint about building `dotnet` inside the `test` block:  
```
==> brew test --verbose dotnet
==> FAILED
==> Testing dotnet
Error: dotnet: failed
An exception occurred within a child process:
  CompilerSelectionError: dotnet cannot be built with any available compilers.
Install Clang or run `brew install gcc`.
```

Reply:
Have you tested it with GCC 11?  I think we should usually try that before using LLVM.  If I recall when I tried to fix this in the Linuxbrew repo, it did work with GCC 11 and the blocker was an incompatibility with `icu4c`, which seems to be fixed now.

Reply:
Haven't tried as I was basing on other Linux distros, e.g.
- Fedora: https://src.fedoraproject.org/rpms/dotnet5.0/blob/rawhide/f/dotnet5.0.spec#_81
- Arch: https://github.com/archlinux/svntogit-community/blob/packages/dotnet-core/trunk/PKGBUILD#L23

Also some `dotnet` docs recommend `clang`/`llvm`, e.g.
https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/linux-requirements.md#toolchain-setup
> Install the following packages for the toolchain:
> 
> - cmake
> - llvm-9
> - clang-9

Reply:
I did see the following comment on https://github.com/dotnet/runtime/pull/53220/files:

> Please note that GNU toolchain support is strictly added for community supported platforms in .NET, where LLVM toolchain is either not available or not as complete. If your platform already has good support for LLVM toolchain, it is recommended (per its official support stauts) to use that instead.

---

So, GCC may work, but the build process may be less maintained as upstream only uses LLVM while GCC is supported by community.

Reply:
These are all good points to know.  I think we had problems with linking when using LLVM before, but we fixed those a while back, along with all the other stuff we fixed for LLVM recently.  Since LLVM is only a build/test dependency anyway, it's not creating a huge dependency footprint.  Given these circumstances I think building with LLVM is the right call.